src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependencyTreeTelemetryService.cs

```diff
@@ -132,20 +133,18 @@ public void ObserveTreeUpdateCompleted(bool hasUnresolvedDependency)
 
             bool ObservedAllRules() => _telemetryStates.All(state => state.Value.ObservedAllRules());
 
-            void InitializeProjectId()
+            async Task InitializeProjectIdAsync()
             {
-#pragma warning disable RS0030 // symbol IProjectGuidService is banned
-                IProjectGuidService projectGuidService =
-                    _project.Services.ExportProvider.GetExportedValueOrDefault<IProjectGuidService>();
+                ISafeProjectGuidService projectGuidService =
```

Comment:
Do you mind moving this to just an import in the constructor?

Reply:
no problem

Reply:
It's not optional so you can delete the null path.

Reply:
I'll add a check for whether the Guid is the default guid

Reply:
This will never return a default GUID, that would be a bug.

Reply:
In fact, that's why this new service exists - to make sure we're never in that situation.

Reply:
https://github.com/dotnet/project-system/blob/bed629b18ba823e49daabfb32f21f777d29fd2db/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/VsSafeProjectGuidService.cs#L39

Reply:
Looks like I should check to see if `Guid.Empty` is returned then